superset-frontend/src/components/ListView/Filters.tsx

```diff
@@ -0,0 +1,205 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useState } from 'react';
+import styled from '@emotion/styled';
+import { withTheme } from 'emotion-theming';
+// @ts-ignore
+import Select from 'react-select';
+
+import { Filter, Filters, FilterValue, InternalFilter } from './types';
+
+interface BaseFilter {
+  Header: string;
+  initialValue: any;
+}
+interface SelectFilterProps extends BaseFilter {
+  onSelect: (selected: any) => any;
+  selects: Filter['selects'];
+  emptyLabel?: string;
+}
+
+const StyledSelect = styled(Select)`
+  display: inline;
+  &.is-focused:not(.is-open) > .Select-control {
+    border: none;
+    box-shadow: none;
+  }
+  .Select-control {
+    display: inline-table;
+    border: none;
+    width: 100px;
+    &:focus,
+    &:hover {
+      border: none;
+      box-shadow: none;
+    }
+
+    .Select-arrow-zone {
+      padding-left: 10px;
+    }
+  }
+  .Select-menu-outer {
+    margin-top: 0;
+    border-bottom-left-radius: 0;
+    border-bottom-left-radius: 0;
+  }
+`;
+
+const FilterContainer = styled.div`
+  display: inline;
+  margin-right: 8px;
+`;
+
+const Title = styled.span`
+  font-weight: bold;
+`;
+
+const CLEAR_SELECT_FILTER_VALUE = 'CLEAR_SELECT_FILTER_VALUE';
+
+function SelectFilter({
+  Header,
+  selects = [],
+  emptyLabel = 'None',
+  initialValue,
+  onSelect,
+}: SelectFilterProps) {
+  const clearFilterSelect = {
+    label: emptyLabel,
+    value: CLEAR_SELECT_FILTER_VALUE,
+  };
+  const options = React.useMemo(() => [clearFilterSelect, ...selects], [
+    emptyLabel,
+    selects,
+  ]);
+
+  const [value, setValue] = useState(
+    typeof initialValue === 'undefined'
+      ? clearFilterSelect.value
+      : initialValue,
+  );
+  const onChange = (selected: { label: string; value: any } | null) => {
+    if (selected === null) return;
+    setValue(selected.value);
+    onSelect(
+      selected.value === CLEAR_SELECT_FILTER_VALUE ? undefined : selected.value,
+    );
+  };
+
+  return (
+    <FilterContainer>
+      <Title>{Header}:</Title>
+      <StyledSelect
+        data-test="filters-select"
+        value={value}
+        options={options}
+        onChange={onChange}
+        clearable={false}
+      />
+    </FilterContainer>
+  );
+}
+
+interface SearchHeaderProps extends BaseFilter {
+  Header: string;
+  onSubmit: (val: string) => void;
+}
+
+const SearchInput = styled.input`
+  background-color: #fff;
+  background-image: none;
+  border: 1px solid #ccc;
```

Comment:
Happy to do it myself, so this is more of a reminder to myself: we should move all these colors and thematic style parameters up into the theme, and keep them in lockstep with variables.less for the time being.